diff --git a/debian/tests/acl-test b/debian/tests/acl-test
new file mode 100644
index 0000000..0d53f2d
--- /dev/null
+++ b/debian/tests/acl-test
@@ -0,0 +1,64 @@
+#!/bin/sh
+
+#
+# autopkgtest check: Tests all of the acl command line tools.
+# For example: Create a image.img file and mount it with 'acl' option
+# to /mnt mount point then get and set access control lists of some files
+# in /mnt directory
+
+set -e
+
+echo "--------------------------------"
+echo "Create a image.img file and mount it with 'acl' option"
+echo "--------------------------------"
+dd if=/dev/zero of=image.img bs=1M count=1024
+mkfs.ext2 image.img
+mount -o acl image.img /mnt
+echo "-----------------------------------"
+echo "Create a image.img file and mount it with 'acl' option  SUCCESS"
+echo "-----------------------------------"
+echo
+
+echo "--------------------------------"
+echo "Create sample1.txt and sample2.txt files for testing"
+echo "--------------------------------"
+touch /mnt/sample1.txt /mnt/sample2.txt
+echo "-----------------------------------"
+echo "Create sample1.txt and sample2.txt files SUCCESS"
+echo "-----------------------------------"
+echo
+
+echo "--------------------------------"
+echo "Get sample1.txt and sample2.txt files access control lists"
+echo "--------------------------------"
+getfacl /mnt/sample1.txt /mnt/sample2.txt
+echo "-----------------------------------"
+echo "Get sample1.txt and sample2.txt files access control lists SUCCESS"
+echo "-----------------------------------"
+echo
+
+echo "--------------------------------"
+echo "Change a minimum access control lists for sample1.txt file"
+echo "--------------------------------"
+chacl u::rwx,g::r-x,o::r-- /mnt/sample1.txt
+getfacl /mnt/sample1.txt /mnt/sample2.txt
+echo "-----------------------------------"
+echo "Change a minimum access control lists for sample1.txt SUCCESS"
+echo "-----------------------------------"
+echo
+
+echo "--------------------------------"
+echo "Copying the access control lists of sample1.txt to sample2.txt"
+echo "--------------------------------"
+getfacl /mnt/sample1.txt | setfacl --set-file=- /mnt/sample2.txt
+getfacl /mnt/sample1.txt /mnt/sample2.txt
+echo "-----------------------------------"
+echo "Copying the access control lists of sample1.txt to sample2.txt SUCCESS"
+echo "-----------------------------------"
+echo
+rm -f /mnt/sample1.txt /mnt/sample2.txt
+umount /mnt
+rm -f image.img
+echo
+echo "acl-test: TEST PASSED"
diff --git a/debian/tests/control b/debian/tests/control
new file mode 100644
index 0000000..9ff035f
--- /dev/null
+++ b/debian/tests/control
@@ -0,0 +1,3 @@
+Tests: acl-test
+Depends: @, mount
+Restrictions: needs-root, allow-stderr
