diff --git a/debian/tests/control b/debian/tests/control
new file mode 100644
index 0000000..3d25185
--- /dev/null
+++ b/debian/tests/control
@@ -0,0 +1,3 @@
+Tests: dcfldd-test
+Depends: @
+Restrictions: allow-stderr
diff --git a/debian/tests/dcfldd-test b/debian/tests/dcfldd-test
new file mode 100644
index 0000000..bc45f72
--- /dev/null
+++ b/debian/tests/dcfldd-test
@@ -0,0 +1,44 @@
+#!/bin/sh
+
+#
+# Copy /dev/zero to driveimage.dd with hashwindow=20k by dcfldd command
+#
+
+set -e
+
+# Create a md5 file for testing
+cat <<EOF> /tmp/md5
+0 - 20480: daa100df6e6711906b61c9ab5aa16032
+20480 - 40960: daa100df6e6711906b61c9ab5aa16032
+40960 - 61440: daa100df6e6711906b61c9ab5aa16032
+61440 - 81920: daa100df6e6711906b61c9ab5aa16032
+81920 - 102400: daa100df6e6711906b61c9ab5aa16032
+
+Total (md5): 4c6426ac7ef186464ecbb0d81cbfcb1e
+EOF
+
+# Create a sha256 file for testing
+cat <<EOF> /tmp/sha256
+0 - 20480: cc61635da46b2c9974335ea37e0b5fd660a5c8a42a89b271fa7ec2ac4b8b26f6
+20480 - 40960: cc61635da46b2c9974335ea37e0b5fd660a5c8a42a89b271fa7ec2ac4b8b26f6
+40960 - 61440: cc61635da46b2c9974335ea37e0b5fd660a5c8a42a89b271fa7ec2ac4b8b26f6
+61440 - 81920: cc61635da46b2c9974335ea37e0b5fd660a5c8a42a89b271fa7ec2ac4b8b26f6
+81920 - 102400: cc61635da46b2c9974335ea37e0b5fd660a5c8a42a89b271fa7ec2ac4b8b26f6
+
+Total (sha256): f627ca4c2c322f15db26152df306bd4f983f0146409b81a4341b9b340c365a16
+EOF
+
+echo "--------------------------------"
+echo "---Testing for dcfldd command---" 
+echo "--------------------------------"
+dcfldd if=/dev/zero hash=md5,sha256 hashwindow=20k md5log=md5.txt sha256log=sha256.txt hashconv=after bs=1k count=100 conv=noerror of=driveimage.dd
+diff /tmp/md5 md5.txt
+diff /tmp/sha256 sha256.txt
+echo "-----------------------------------"
+echo "Testing for dcfldd command SUCCESS"
+echo "-----------------------------------"
+echo
+
+rm -f /tmp/md5 /tmp/sha256 md5.txt sha256.txt driveimage.dd
+echo
+echo "dcfldd-test: TEST PASSED"
