diff --git a/dosfstools-4.1/debian/tests/control b/dosfstools-4.1/debian/tests/control
new file mode 100644
index 0000000..4473fce
--- /dev/null
+++ b/dosfstools-4.1/debian/tests/control
@@ -0,0 +1,3 @@
+Tests: dosfstools.sh
+Depends: @
+Restrictions: needs-root, allow-stderr
diff --git a/dosfstools-4.1/debian/tests/dosfstools.sh b/dosfstools-4.1/debian/tests/dosfstools.sh
new file mode 100644
index 0000000..ae31663
--- /dev/null
+++ b/dosfstools-4.1/debian/tests/dosfstools.sh
@@ -0,0 +1,47 @@
+#!/bin/sh
+#
+#===============================================================
+# DESCRIPTION: Testing for commands of dosfstools package
+#              mkfs.fat: create an MS-DOS filesystem under Linux
+#              fsck.fat: check and repair MS-DOS filesystems
+#              fatlabel: set or get a MS-DOS filesystem label
+#              from a given device
+#===============================================================
+
+set -e
+label="TEST LABEL"
+img_file="${PWD}/image.img"
+dd if=/dev/zero of=$img_file bs=1k count=1024
+
+# /sbin/mkfs.vfat, /sbin/mkfs.msdos, /sbin/mkdosfs are just a symlink to /sbin/mkfs.fat
+#  mkfs.fat: create an MS-DOS filesystem under Linux
+mkfs.fat $img_file
+if [ $? != 0 ]; then
+        echo "mkfs.fat: failed!"
+fi
+
+# /sbin/dosfsck, /sbin/fsck.msdos, /sbin/fsck.vfat are just a symlink to /sbin/fsck.fat
+# fsck.fat: check and repair MS-DOS filesystems
+fsck.fat $img_file
+if [ $? != 0 ]; then
+        echo "fsck.fat: failed!"
+fi
+
+# /sbin/dosfslabel is just a symlink to /sbin/fatlabel
+# fatlabel: set or get a MS-DOS filesystem label from a given device
+fatlabel $img_file "$label"
+if [ $? != 0 ]; then
+        echo "fatlabel: cannot set label!"
+fi
+
+# get label return label with trailing spaces, strip them
+actual_label=`fatlabel $img_file | sed 's/[ ]*$//'`
+expect_label=`echo "$label" | sed 's/[ ]*$//'`
+if [ "$actual_label" != "$expect_label" ]; then
+        echo "fatlabel: label set not correct!"
+fi
+
+# remove non-necesscery file
+rm -f $img_file
+
+echo "=====DONE====="
-- 
2.16.1
