diff --git a/findutils-4.6.0+git+20170828/debian/tests/control b/findutils-4.6.0+git+20170828/debian/tests/control
new file mode 100644
index 0000000..762161e
--- /dev/null
+++ b/findutils-4.6.0+git+20170828/debian/tests/control
@@ -0,0 +1,2 @@
+Tests: findutils_script
+Depends: @, diffutils
diff --git a/findutils-4.6.0+git+20170828/debian/tests/findutils_script b/findutils-4.6.0+git+20170828/debian/tests/findutils_script
new file mode 100644
index 0000000..4f345b2
--- /dev/null
+++ b/findutils-4.6.0+git+20170828/debian/tests/findutils_script
@@ -0,0 +1,32 @@
+#!/bin/sh
+
+set -e
+
+#Package name here
+pkg=findutils
+
+if [ "$ADTTMP" = "" ] ; then
+  ADTTMP=`mktemp -d /tmp/${pkg}-test.XXXXXX`
+fi
+
+cd $ADTTMP
+
+echo "====Start $pkg===="
+
+execute_command_result(){
+  if [ "$exp_output" = "$output" ]; then
+    echo "PASS"
+  else
+    exit 1
+  fi
+}
+
+exp_output="/usr/bin/find"
+output=$(find /usr/bin/ -name "find")
+execute_command_result
+
+exp_output=$(ls /)
+output=$(echo "/" | xargs ls)
+execute_command_result
+
+echo "====Done===="
-- 
2.15.1

