diff --git a/cpio-2.12+dfsg/debian/tests/control b/cpio-2.12+dfsg/debian/tests/control
new file mode 100644
index 0000000..d0d295f
--- /dev/null
+++ b/cpio-2.12+dfsg/debian/tests/control
@@ -0,0 +1,2 @@
+Tests: cpio_script
+Depends: @
diff --git a/cpio-2.12+dfsg/debian/tests/cpio_script b/cpio-2.12+dfsg/debian/tests/cpio_script
new file mode 100644
index 0000000..1d20ffa
--- /dev/null
+++ b/cpio-2.12+dfsg/debian/tests/cpio_script
@@ -0,0 +1,57 @@
+#!/bin/sh
+
+set -e
+
+pkg=cpio
+
+if [ "$ADTTMP" = "" ] ; then
+  ADTTMP=`mktemp -d /tmp/${pkg}-test.XXXXXX`
+fi
+
+cd $ADTTMP
+
+echo "====Start $pkg===="
+
+log_file=$pkg".log"
+check=0
+
+# Create files for testing
+mkdir -p ${PWD}/test_cpio >> $log_file 2>&1
+cd ${PWD}/test_cpio
+touch tmp1.txt tmp2.txt >> $log_file 2>&1
+
+# Create the archive file
+{ ls | cpio -o > tmp.cpio; } >> $log_file 2>&1
+
+if [ $? -ne 0 ]; then
+  check=1
+fi
+
+ls | grep "tmp.cpio" >> $log_file 2>&1
+
+if [ $? -ne 0 ]; then
+  check=1
+fi
+
+# Extract the archived file
+rm tmp1.txt tmp2.txt >> $log_file 2>&1
+
+cpio -i < tmp.cpio >> $log_file 2>&1
+
+if [ $? -ne 0 ]; then
+  check=1
+fi
+
+count=`ls | grep "tmp[0-9].txt" | wc -l`
+
+if [ $count -ne 2 ]; then
+  check=1
+fi
+
+if [ $check -eq 0 ]; then
+  echo "PASS"
+else
+  exit 1
+fi
+
+echo "====Done===="
-- 
2.15.1

