diff --git a/debian/tests/bison-test b/debian/tests/bison-test
new file mode 100644
index 0000000..c9bd4a6
--- /dev/null
+++ b/debian/tests/bison-test
@@ -0,0 +1,79 @@
+#!/bin/sh
+
+#
+# Testing for bison and bison.yacc commands.
+#
+
+set -e
+
+# Create a sample.ypp file for testing
+cat <<EOF> /tmp/sample.ypp
+%{
+  int vbltable[26];
+%}
+
+
+%token  NAME 
+%token  NUMBER
+%token  EQ
+%token PLUS MINUS TIMES DIV
+%left MINUS PLUS
+%left TIMES DIV 
+%nonassoc UMINUS
+
+%%
+
+statement_list
+        : statement 
+        | statement statement_list
+        
+statement
+        : NAME EQ expression ';' {vbltable[$1] = $3; }
+        
+expression
+        : expression PLUS expression {$$ = $1 + $3;}
+        | expression MINUS expression {$$ = $1 - $3;}
+        | expression TIMES expression {$$ = $1 * $3;}
+        | expression DIV   expression {$$ = $1 / $3;}
+        | MINUS expression %prec UMINUS {$$ = - $2;}    
+        | '(' expression ')' { $$ = $2; } 
+        | NUMBER
+        | NAME   { $$ = vbltable[$1]; }
+        
+%%
+main()
+{
+   int i;
+
+ for (i = 0; i < 26; i++)
+   vbltable[i] = 0;
+ yyparse();
+ for (i = 0; i < 26; i++)
+  if (vbltable[i] != 0)
+   printf("Table: %d :%d\n",  i, vbltable[i]);
+
+}
+EOF
+
+echo "--------------------------------"
+echo "Testing for bison command" 
+echo "--------------------------------"
+bison /tmp/sample.ypp
+echo "-----------------------------------"
+echo "Testing for bison command SUCCESS"
+echo "-----------------------------------"
+echo
+
+echo "--------------------------------"
+echo "Testing for bison.yacc command" 
+echo "--------------------------------"
+bison.yacc /tmp/sample.ypp
+echo "-----------------------------------"
+echo "Testing for bison.yacc command SUCCESS"
+echo "-----------------------------------"
+echo
+
+rm -f /tmp/sample.ypp
+rm -f sample.tab.cpp y.tab.c
+echo
+echo "bison-test: TEST PASSED"
diff --git a/debian/tests/control b/debian/tests/control
new file mode 100644
index 0000000..2896c15
--- /dev/null
+++ b/debian/tests/control
@@ -0,0 +1,2 @@
+Tests: bison-test
+Depends: @
