diff --git a/flex-2.6.1/debian/tests/control b/flex-2.6.1/debian/tests/control
new file mode 100644
index 0000000..3a1789e
--- /dev/null
+++ b/flex-2.6.1/debian/tests/control
@@ -0,0 +1,2 @@
+Tests: flex_script
+Depends: @
diff --git a/flex-2.6.1/debian/tests/flex_script b/flex-2.6.1/debian/tests/flex_script
new file mode 100644
index 0000000..c0fa330
--- /dev/null
+++ b/flex-2.6.1/debian/tests/flex_script
@@ -0,0 +1,39 @@
+#!/bin/sh
+
+set -e
+
+#Package name here
+pkg=flex
+sample_file=sample.lex
+
+if [ "$ADTTMP" = "" ] ; then
+  ADTTMP=`mktemp -d /tmp/${pkg}-test.XXXXXX`
+fi
+
+cp debian/tests/testing-data/$sample_file $ADTTMP/$sample_file
+cd $ADTTMP
+
+echo "====Start $pkg===="
+
+log_file=$pkg".log"
+exp_file_flex=lex.yy.c
+exp_file_flex_plus=lex.yy.cc
+
+command_execute_result(){
+  if [ -s $log_file ]; then
+    echo "PASS"
+    rm $log_file
+  else
+    exit 1
+  fi
+}
+
+flex $sample_file
+find -name $exp_file_flex > $log_file
+command_execute_result
+
+flex++ $sample_file
+find -name $exp_file_flex_plus > $log_file
+command_execute_result
+
+echo "====Done===="
diff --git a/flex-2.6.1/debian/tests/testing-data/sample.lex b/flex-2.6.1/debian/tests/testing-data/sample.lex
new file mode 100644
index 0000000..8438ef4
--- /dev/null
+++ b/flex-2.6.1/debian/tests/testing-data/sample.lex
@@ -0,0 +1,26 @@
+/* 
+ * Sample: 
+ * Description: Replace the string "username" from standard input 
+ *              with the user's login name (e.g. lgao)
+ * Usage: (1) $ flex sample1.lex
+ *        (2) $ gcc lex.yy.c -lfl
+ *        (3) $ ./a.out
+ *            stdin> username
+ *	      stdin> Ctrl-D
+ * Question: What is the purpose of '%{' and '%}'?
+ *           What else could be included in this section?
+ */
+
+%{
+/* need this for the call to getlogin() below */
+#include <unistd.h>
+%}
+
+%%
+username	printf("%s\n", getlogin());
+%%
+
+main()
+{
+  yylex();
+}
-- 
2.15.1

