From 5e81913578e6f90f59c12fcb86685a88f572e75d Mon Sep 17 00:00:00 2001
From: Tran Van Tam <tam.tranvan@toshiba-tsdv.com>
Date: Thu, 16 Mar 2017 11:59:33 +0000
Subject: [PATCH] Fix refresh window but still allow fullscreen resize.

Signed-off-by: Tran Van Tam <tam.tranvan@toshiba-tsdv.com>
---
 src/wmmgr.cc | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/src/wmmgr.cc b/src/wmmgr.cc
index f23975e..cc57c2c 100644
--- a/src/wmmgr.cc
+++ b/src/wmmgr.cc
@@ -2291,8 +2291,8 @@ void YWindowManager::relocateWindows(long workspace, int screen, int dx, int dy)
 
 void YWindowManager::resizeWindows() {
     for (YFrameWindow * f = topLayer(); f; f = f->nextLayer()) {
-        if (f->inWorkArea()) {
-            if (f->isMaximized())
+        if (f->inWorkArea() || f->isMaximized() || f->isFullscreen()) {
+            if (f->isMaximized() || f->isFullscreen())
                 f->updateDerivedSize(WinStateMaximizedVert | WinStateMaximizedHoriz);
                 f->updateLayout();
         }
@@ -3431,7 +3431,7 @@ void YWindowManager::UpdateScreenSize(XEvent *event) {
     updateXineramaInfo(nw, nh);
 
     if (width() != nw ||
-        height() != nh)
+        height() != nh || 1)
     {
 
         MSG(("xrandr: %d %d",
-- 
2.1.4

