diff --git a/autogen.sh b/autogen.sh
index 34bb633..fa1e821 100755
--- a/autogen.sh
+++ b/autogen.sh
@@ -10,7 +10,7 @@ if [ -x "`which git 2>/dev/null`" -a -d .git ]; then
 	DATE=$(git log --date=iso|grep '^Date:'|head -1|awk '{print$2}')
 	VERSION=$(git describe --tags|sed 's,^[^0-9]*,,;s,[-_],.,g;s,\.g.*$,,')
 	(
-	   echo -e "# created with git log -n 200 --abbrev-commit --decorate --stat=76 -M -C|fmt -sct -w80\n"
+	   echo "# created with git log -n 200 --abbrev-commit --decorate --stat=76 -M -C|fmt -sct -w80\n"
 	   git log -n 200 --abbrev-commit --decorate --stat=76 -M -C|fmt -sct -w80
 	)>ChangeLog
 	(
@@ -46,7 +46,7 @@ subst="s:%%PACKAGE%%:$PACKAGE:g
 sed -r -e "$subst" icewm.spec.in >icewm.spec
 sed -r -e "$subst" icewm.lsm.in >icewm.lsm
 
-echo -e "PACKAGE=$PACKAGE\nVERSION=$VERSION" >VERSION
+echo "PACKAGE=$PACKAGE\nVERSION=$VERSION" >VERSION
 
 autoreconf -iv
 
