diff --git a/debian/tests/bzip2-test b/debian/tests/bzip2-test
new file mode 100644
index 0000000..fe1bf85
--- /dev/null
+++ b/debian/tests/bzip2-test
@@ -0,0 +1,40 @@
+#!/bin/sh
+
+#
+# Create sample.txt for testing then compress and decompress using
+# bzip2 command with -z and -d options
+#
+
+set -e
+
+echo "--------------------------------"
+echo "Create sample.txt file for testing"
+echo "--------------------------------"
+touch sample.txt
+echo "-----------------------------------"
+echo "Create sample.txt file for testing SUCCESS"
+echo "-----------------------------------"
+echo
+
+echo "--------------------------------"
+echo "---Testing for bzip2 command ---"
+echo "--------------------------------"
+echo "Compress sample.txt file to sample.txt.bz2 file with -z option"
+bzip2 -z sample.txt
+echo "-----------------------------------"
+echo "Compress sample.txt file to sample.txt.bz2 file with -z option SUCCESS"
+echo "-----------------------------------"
+echo
+
+echo "--------------------------------"
+echo "Decompress sample.txt.bz2 file to sample.txt file with -d option"
+echo "--------------------------------"
+bzip2 -d sample.txt.bz2
+echo "-----------------------------------"
+echo "Decompress sample.txt.bz2 file to sample.txt file with -d option SUCCESS"
+echo "-----------------------------------"
+echo
+
+rm -f sample.txt
+echo
+echo "bzip2-test: TEST PASSED"
diff --git a/debian/tests/control b/debian/tests/control
new file mode 100644
index 0000000..bafced6
--- /dev/null
+++ b/debian/tests/control
@@ -0,0 +1,2 @@
+Tests: bzip2-test
+Depends: @
