diff --git a/dnsmasq-2.78/debian/tests/control b/dnsmasq-2.78/debian/tests/control
new file mode 100644
index 0000000..61e94c9
--- /dev/null
+++ b/dnsmasq-2.78/debian/tests/control
@@ -0,0 +1,3 @@
+Tests: dnsmasq-test
+Depends: @
+Restrictions: needs-root, allow-stderr
diff --git a/dnsmasq-2.78/debian/tests/dnsmasq-test b/dnsmasq-2.78/debian/tests/dnsmasq-test
new file mode 100644
index 0000000..9eda7fa
--- /dev/null
+++ b/dnsmasq-2.78/debian/tests/dnsmasq-test
@@ -0,0 +1,51 @@
+#!/bin/sh
+
+#
+# Testing for dnsmasq command.
+# Start, stop and restart dnsmasq service.
+# Read and syntax check configuration file(s).
+#
+
+set -e 
+
+echo "--------------------------------"
+echo "---Testing for dnsmasq command--"
+echo "--------------------------------"
+echo
+
+echo "--------------------------------"
+echo "Start dnsmasq service"
+echo "--------------------------------"
+service dnsmasq start
+echo "-----------------------------------"
+echo "Start dnsmasq service SUCCESS"
+echo "-----------------------------------"
+echo
+
+echo "--------------------------------"
+echo "Stop dnsmasq service"
+echo "--------------------------------"
+service dnsmasq stop
+echo "-----------------------------------"
+echo "Stop dnsmasq service SUCCESS"
+echo "-----------------------------------"
+echo
+
+echo "--------------------------------"
+echo "Restart dnsmasq service"
+echo "--------------------------------"
+service dnsmasq restart
+echo "-----------------------------------"
+echo "Restart dnsmasq service SUCCESS"
+echo "-----------------------------------"
+echo
+
+echo "--------------------------------"
+echo "Read and syntax check configuration file(s)"
+echo "--------------------------------"
+dnsmasq --test
+echo "-----------------------------------"
+echo "Read and syntax check configuration file(s) SUCCESS"
+echo "-----------------------------------"
+echo
-- 
2.15.1

