diff --git a/debian/tests/control b/debian/tests/control
new file mode 100644
index 0000000..78bd765
--- /dev/null
+++ b/debian/tests/control
@@ -0,0 +1,3 @@
+Tests: ed-test
+Depends: @
diff --git a/debian/tests/ed-test b/debian/tests/ed-test
new file mode 100644
index 0000000..9bfe50d
--- /dev/null
+++ b/debian/tests/ed-test
@@ -0,0 +1,47 @@
+#!/bin/sh
+
+#
+# Replace, delete and move content of file using ed command
+#
+
+set -e
+
+# Create a ed.txt file for testing
+cat <<EOF> /tmp/ed.txt
+Hello ed
+Line 2
+Line 3
+Line 4
+Line 5
+EOF
+
+# Create a ed.expected file for testing
+cat <<EOF> /tmp/ed.expected
+Hi ed
+Line 5
+Line 3
+Line 4
+EOF
+
+echo "--------------------------------"
+echo "-----Testing for ed command-----" 
+echo "--------------------------------"
+# Replace "Hello" with "Hi"
+echo ",s/Hello/Hi/g\nw" | ed -s /tmp/ed.txt
+
+# Move line 3,4 to the end of file
+echo "3,4m$\nw" | ed -s /tmp/ed.txt
+
+# Delete line number 2
+echo "2d\nw" | ed -s /tmp/ed.txt
+
+# Check content of "ed.txt"
+diff /tmp/ed.expected /tmp/ed.txt
+echo "------------------------------"
+echo "Testing for ed command SUCCESS"
+echo "------------------------------"
+echo
+
+rm -f /tmp/ed.txt /tmp/ed.expected
+echo
+echo "ed-test: TEST PASSED"
