diff --git a/debian/tests/bc-test b/debian/tests/bc-test
new file mode 100644
index 0000000..a1c0e92
--- /dev/null
+++ b/debian/tests/bc-test
@@ -0,0 +1,31 @@
+#!/bin/sh
+
+#
+# Test some calculations with bc command.
+#
+
+set -e
+
+# Create output_expected.txt file for testing
+cat <<EOF> /tmp/output_expected.txt
+140
+11
+0
+EOF
+
+echo "--------------------------------"
+echo "Test bc command with some calculations"
+echo "--------------------------------"
+sed -e 's/\s*\(.*\).*/\1/' << EOF | bc >> /tmp/bc-result.txt
+121+19
+sqrt(121)
+50>150
+EOF
+diff /tmp/output_expected.txt /tmp/bc-result.txt
+echo "-----------------------------------"
+echo "Test bc command with some calculations SUCCESS"
+echo "-----------------------------------"
+echo
+rm -f /tmp/output_expected.txt /tmp/bc-result.txt
+echo
+echo "bc-test: TEST PASSED"
diff --git a/debian/tests/control b/debian/tests/control
new file mode 100644
index 0000000..b1653ec
--- /dev/null
+++ b/debian/tests/control
@@ -0,0 +1,2 @@
+Tests: bc-test
+Depends: @
