diff --git a/gawk-4.1.4+dfsg/debian/tests/control b/gawk-4.1.4+dfsg/debian/tests/control
new file mode 100644
index 0000000..b8a6e1f
--- /dev/null
+++ b/gawk-4.1.4+dfsg/debian/tests/control
@@ -0,0 +1,2 @@
+Tests: gawk_script
+Depends: @
diff --git a/gawk-4.1.4+dfsg/debian/tests/gawk_script b/gawk-4.1.4+dfsg/debian/tests/gawk_script
new file mode 100644
index 0000000..eb4dfc6
--- /dev/null
+++ b/gawk-4.1.4+dfsg/debian/tests/gawk_script
@@ -0,0 +1,49 @@
+#!/bin/sh
+
+set -e
+
+#Package name here
+pkg=gawk
+
+if [ "$ADTTMP" = "" ] ; then
+  ADTTMP=`mktemp -d /tmp/${pkg}-test.XXXXXX`
+fi
+
+cd $ADTTMP
+
+echo "====Start $pkg===="
+
+log_file=$pkg".log"
+exp_output="Hello gawk"
+gawk_sample_file=gawk.awk
+igawk_sample_file=iigawk.awk
+
+check_command_execute_result(){
+  if [ "$output" = "$exp_output" ]; then
+    echo "PASS"
+  else
+    exit 1
+  fi
+}
+
+cat << EOF > $gawk_sample_file
+BEGIN {
+print "$exp_output"
+}
+EOF
+ 
+output=$(gawk -f $gawk_sample_file)
+check_command_execute_result
+
+cat << EOF > $igawk_sample_file
+@include getopt.awk
+
+BEGIN {
+print "$exp_output"
+}
+EOF
+
+output=$(igawk -f $igawk_sample_file)
+check_command_execute_result
+
+echo "====Done===="
-- 
2.15.1

